java/impl/gcp/src/main/java/co/worklytics/psoxy/SecretManagerConfigService.java

```diff
@@ -71,6 +72,25 @@ public void putConfigProperty(ConfigProperty property, String value) {
                 SecretVersion version = client.addSecretVersion(secretName, payload);
 
                 log.info(String.format("Property: %s, stored version %s", secretName, version.getName()));
+
+                SecretManagerServiceClient.ListSecretVersionsPage enabledVersions = client.listSecretVersions(ListSecretVersionsRequest.newBuilder()
+                                .setFilter("state:ENABLED")
+                                .setParent(secretName.toString())
+                                // Reduce the page, as each version will be disabled one by one
+                                .setPageSize(20)
```

Comment:
Just the first page to avoid doing too much in a single request

Reply:
although affected customer won't require this, I think worth happening in case a DISABLE request fails or something in future; this gives some robustness against versions being left behind and gradually increasing costs